backtrace-library/src/main/java/backtraceio/library/base/BacktraceBase.java

```diff
@@ -445,6 +479,72 @@ public boolean addBreadcrumb(String message, Map<String, Object> attributes, Bac
         return database.getBreadcrumbs().addBreadcrumb(message, attributes, type, level);
     }
 
+    /**
+     * Enable backtrace metrics support
+     * @param settings  Backtrace metrics settings
+     */
+    public void enableMetrics(BacktraceMetricsSettings settings) {
+        // For tracking crash-free sessions we need to add
+        // application.session and application.version to Backtrace attributes
+        String sessionId = UUID.randomUUID().toString();
+        this.attributes.put("application.session", sessionId);
+        try {
+            this.attributes.put("application.version", this.context.getPackageManager()
+                    .getPackageInfo(this.context.getPackageName(), 0).versionName);
+        } catch (PackageManager.NameNotFoundException e) {
+            BacktraceLogger.e(LOG_TAG, "Could not resolve package version information for Backtrace metrics");
+        }
+
+        metrics = new BacktraceMetrics(context, attributes, settings);
+        metrics.startMetricsEventHandlers(backtraceApi);
+        metrics.sendStartupEvent();
+    }
+
+    public boolean addUniqueEvent(String attributeName) {
+        return metrics.addUniqueEvent(attributeName);
+    }
+
+    public boolean addUniqueEvent(String attributeName, Map<String, Object> attributes) {
+        return metrics.addUniqueEvent(attributeName, attributes);
+    }
+
+    public boolean addSummedEvent(String metricGroupName) {
+        return metrics.addSummedEvent(metricGroupName);
+    }
+
+    public boolean addSummedEvent(String metricGroupName, Map<String, Object> attributes) {
+        return metrics.addSummedEvent(metricGroupName, attributes);
+    }
+
+    public ConcurrentLinkedDeque<UniqueEvent> getUniqueEvents() {
+        return metrics.getUniqueEvents();
+    }
+
+    public ConcurrentLinkedDeque<SummedEvent> getSummedEvents() {
```

Comment:
same https://www.geeksforgeeks.org/how-to-make-a-collection-read-only-in-java/

Reply:
In @konraddysput implementation we can modify events, I did the same:
https://github.com/backtrace-labs/backtrace-unity/blob/553aab2b39c318ff96ebed4bc739bf2c87304649/Runtime/Services/BacktraceMetrics.cs#L154